hydra_provider/src/data_feed_service/random/service.rs

```diff
@@ -0,0 +1,51 @@
+use std::sync::atomic::{AtomicBool, Ordering};
+use std::sync::Arc;
+use tonic::{async_trait, Request, Response, Status};
+use log::info;
+
+use crate::database_service::DatabaseService;
+use crate::data_feed_service::message::HydraInput;
+use crate::data_feed_service::random::iterator::RandomDataFeedIterator;
+
+use crate::data_feed_service::proto::random_data_feed_server::*;
+use crate::data_feed_service::proto::Empty;
+
+pub struct RandomDataFeedService<'a> {
+	iterator: RandomDataFeedIterator,
+	database_service: &'a DatabaseService,
```

Comment:
Perhaps we might have less headaches by using Arc<DatabaseService> instead of a lifetime reference

Reply:
oh I see